extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -362,4 +373,50 @@ private synchronized void setPatchMatchingPolicyEnabled() {
 
         protected abstract boolean httpPermissionsEmpty();
     }
+
+    public void setMtlsCertificateRoleProperties(HttpConfiguration config) {
+        if (config.auth.certificateRoleProperties.isPresent()) {
+            Map<String, Set<String>> roles = new HashMap<>();
+
+            Path rolesPath = config.auth.certificateRoleProperties.get();
+            URL rolesResource = null;
+            if (Files.exists(rolesPath)) {
+                try {
+                    rolesResource = rolesPath.toUri().toURL();
+                } catch (MalformedURLException e) {
+                    // The Files.exists(rolesPath) check has succeeded therefore this exception can't happen in this case
+                }
+            } else {
+                rolesResource = Thread.currentThread().getContextClassLoader().getResource(rolesPath.toString());
+            }
+            if (rolesResource == null) {
+                throw new ConfigurationException(
+                        "quarkus.http.auth.certificate-role-properties location can not be resolved",
+                        Set.of("quarkus.http.auth.certificate-role-properties"));
+            }
+            Properties rolesProps = new Properties();
+            try (InputStream is = rolesResource.openStream()) {
+                rolesProps.load(is);
+                for (Map.Entry<Object, Object> e : rolesProps.entrySet()) {
+                    log.debugf("Added role mapping for %s:%s", e.getKey(), e.getValue());
+                    roles.put((String) e.getKey(), parseRoles((String) e.getValue()));
+                }
+            } catch (Exception e) {
+                log.warnf("Unable to read roles mappings from %s:%s", rolesPath, e.getMessage());
+            }
+            InstanceHandle<MtlsAuthenticationMechanism> mtls = Arc.container().instance(MtlsAuthenticationMechanism.class);
+            if (mtls.isAvailable()) {
+                mtls.get().setRoleMappings(roles);
+            }
+        }
+
+    }
+
+    private static Set<String> parseRoles(String value) {
+        Set<String> roles = new HashSet<>();
+        for (String s : value.split(",")) {
+            roles.add(s.trim());
+        }
+        return roles;
```

Comment:
It might be a good idea to either do this stream-wise or return `Set.copyOf(roles)` just so that the long-lived set is as small as possible. Obviously this would only make a difference if there were really a lot of roles though, so it's up to you.